### PR TITLE
#100 add new lifecycle properties for making service registration optional

### PR DESCRIPTION
**Pull request by aatarasoff**

### CODE REVIEW

File: spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulLifecycle.java [hunk 1]
```diff
@@ -170,7 +173,7 @@ private void deregister(String serviceId) {
 
 	@Override
 	protected boolean isEnabled() {
-		return properties.isEnabled();
+		return lifecycleProperties.isEnabled();
```
Comment (spencergibb) [inline spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulLifecycle.java]: How about `properties.isEnabled() && lifecycleProperties.isEnabled()`? That way if `spring.cloud.consul.enabled=false` the lifecycle isn't started as well.

Comment (aatarasoff) [inline spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulLifecycle.java]: As I see if `spring.cloud.enabled=false` or `spring.cloud.consul.discovery.enabled=false` then  `ConsulDiscoveryClientConfiguration` will not enabled according to `@ConditionalOnConsulEnabled` and `@ConditionalOnProperty` annotations and then `ConsulLifecycle` will be eliminated despite the value of `spring.cloud.consul.discovery.lifecycle`.

Comment (spencergibb) [inline spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulLifecycle.java]: good point.

File: spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/LifecycleProperties.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.Data;
+import lombok.extern.apachecommons.CommonsLog;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * @author Aleksandr Tarasov (aatarasoff)
+ */
+@ConfigurationProperties(prefix = "spring.cloud.consul.discovery.lifecycle")
+@Data
+@CommonsLog
```
Comment (spencergibb) [inline spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/LifecycleProperties.java]: not needed

File: spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/LifecycleProperties.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+public class LifecycleProperties {
+    private boolean enabled = true;
```
Comment (spencergibb) [inline spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/LifecycleProperties.java]: we use tabs, not spaces.

File: spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/LifecycleProperties.java (hunk 2 above)
Comment (aatarasoff) [inline spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/LifecycleProperties.java]: removed

File: spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/LifecycleProperties.java (hunk 3 above)
Comment (aatarasoff) [inline spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/LifecycleProperties.java]: style is fixed

**Comment by spencergibb** [comment]
Squashed and merged via 371e7ce87ae7c535050c93c1a9a2973679f99e0e

**Comment by spencergibb** [comment]
Shoot, forgot to ask if you signed the [contributor's agreement](https://support.springsource.com/spring_committer_signup)?

**Comment by spencergibb** [comment]
He did, see #103

**Pull request closed without merging**